js/apps/system/_admin/aardvark/APP/frontend/js/views/collectionsView.js

```diff
@@ -433,21 +433,32 @@
             collName: collName,
             wfs: wfs,
             isSystem: isSystem,
-            collType: collType,
-            shards: shards,
-            shardKeys: shardKeys
+            collType: collType
           };
 
-          if (smartJoinAttribute !== '') {
-            tmpObj.smartJoinAttribute = smartJoinAttribute;
-          }
+          if (!isOneShardDB) {
+            if (smartJoinAttribute !== '') {
+              tmpObj.smartJoinAttribute = smartJoinAttribute;
+            }
 
-          tmpObj.distributeShardsLike = distributeShardsLike;
-          if (distributeShardsLike === '' && window.App.isCluster) {
-            // if we are in the cluster and are not using distribute shards like
-            // then we want to make use of the replication factor
-            tmpObj.replicationFactor = replicationFactor === "satellite" ? replicationFactor : Number(replicationFactor);
-            tmpObj.writeConcern = Number(writeConcern);
+            // If we are in a oneShardDB we are not allowed to set those values
+            // They are always inferred
+            if (window.App.isCluster) {
+              tmpObj.shardKeys = shardKeys;
+              if (distributeShardsLike === '') {
+                // if we are not using distribute shards like
+                // then we want to make use of the given shard information
+                tmpObj.shards = shards;
+                tmpObj.replicationFactor = replicationFactor === "satellite" ? replicationFactor : Number(replicationFactor);
+                tmpObj.writeConcern = Number(writeConcern);
+              } else {
+                // If we use distribute shards like on purpose do not add other
+                // sharding information. All of it will be deferred.
+                tmpObj.distributeShardsLike = distributeShardsLike;
+              }
+            } else {
+              tmpObj.shards = shards;
```

Comment:
Why are we assigning something to the `shards` attribute here in the non-cluster case?
It will populate the `numberOfShards` attribute in the end, but for what would that be needed when not running in the cluster?

Reply:
Why should our backend care about `numberOfShards` in a single server? If it does, it would be a severe bug.

Reply:
The backend API requires it as of now.

Reply:
Why? A single server shard not care about `numberOfShards`.

Reply:
I don't believe it.

Reply:
Ah yes, I probably confused the error message for `shardKeys` with the `shards` parameter. Thanks for pointing this out.

Reply:
@jsteemann So ok now?